tests/manage/cluster/nodes/test_disk_failures.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import node
+from ocs_ci.framework.testlib import tier4, ignore_leftovers
+from ocs_ci.framework import config
+from ocs_ci.utility import aws
+from tests.manage.cluster.nodes.base_node_ops import BaseNodes
+
+logger = logging.getLogger(__name__)
+
+
+@tier4
+@pytest.mark.skipif(
+    condition=config.ENV_DATA['platform'] != 'AWS',
+    reason="Tests are not running on AWS deployed cluster"
+)
+@ignore_leftovers
+class TestDetachAttachWorkerVolumeAWS(BaseNodes):
+    """
+    Test class for detach and attach worker volume
+
+    """
+    workers = list()
+    volume = None
+    ec2_volume = None
+
+    def set_workers(self, num_of_workers):
+        """
+        Set the worker nodes for the test
+
+        Args:
+            num_of_workers (int): Number of workers needed for the test
+
+        """
+        nodes = node.get_node_objs()
+
+        # Get all cluster worker nodes
+        workers = [
+            n for n in nodes if 'worker' in n.get('metadata')
+            .get('annotations').get('machine.openshift.io/machine')
+        ]
+
+        # Set the worker nodes for the test, according to the requested
+        # number of node workers
+        self.workers = workers[:num_of_workers]
+
+    def set_volume(self, aws_obj, instance_id):
+        """
+        Initialize volume variables
+
+        Args:
+            aws_obj (AWS): An instance of AWS
+            instance_id (str): The ID of the ec2 instance
+
+        """
+        # Get the data volumes that will be detached and attached later
+        self.volume = aws.get_data_volumes(instance_id)[0]
+
+        # Initialize an ec2 Volume object
+        self.ec2_volume = aws_obj.ec2_resource.Volume(self.volume['id'])
+
+    @pytest.mark.polarion_id("OCS-1085")
+    def test_detach_attach_worker_volume(self, aws_obj, resources):
+        """
+        Detach and attach worker volume
+
+        - Detach the data volume from one of the worker nodes
+        - Validate cluster functionality, without checking cluster and Ceph
+          health (as one node volume is detached, the cluster will be unhealthy)
+          by creating resources and running IO
+        - Attach back the volume to the node
+        - Restart the node so the volume will get re-mounted
+
+        """
+        # Requesting 1 worker node for the test as this case includes detach and
+        # attach of data volume of 1 worker node
+        self.set_workers(1)
+        worker = self.workers[0]
+
+        # Get the worker node's ec2 instance ID and name
+        instance = aws.get_instances_ids_and_names([worker])
+        instance_id = [*instance][0]
+
+        self.set_volume(aws_obj, instance_id)
```

Comment:
same - if you need the volume to detach, a function that will return it will be much easier approach

Reply:
done